src/notation/internal/notationinteraction.cpp

```diff
@@ -731,6 +731,107 @@ void NotationInteraction::selectTopOrBottomOfChord(MoveDirection d)
     showItem(target);
 }
 
+static bool elementLower(const EngravingItem* e1, const EngravingItem* e2)
+{
+    if (!e1->selectable()) {
+        return false;
+    }
+    if (!e2->selectable()) {
+        return true;
+    }
+    if (e1->isNote() && e2->isStem()) {
+        return true;
+    }
+    if (e2->isNote() && e1->isStem()) {
+        return false;
+    }
+    if (e1->z() == e2->z()) {
+        // same stacking order, prefer non-hidden elements
+        if (e1->type() == e2->type()) {
+            if (e1->type() == ElementType::NOTEDOT) {
+                const NoteDot* n1 = static_cast<const NoteDot*>(e1);
+                const NoteDot* n2 = static_cast<const NoteDot*>(e2);
+                if (n1->note() && n1->note()->hidden()) {
+                    return false;
+                } else if (n2->note() && n2->note()->hidden()) {
+                    return true;
+                }
+            } else if (e1->type() == ElementType::NOTE) {
+                const Note* n1 = static_cast<const Note*>(e1);
+                const Note* n2 = static_cast<const Note*>(e2);
+                if (n1->hidden()) {
+                    return false;
+                } else if (n2->hidden()) {
+                    return true;
+                }
+            }
+        }
+        // different types, or same type but nothing hidden - use track
+        return e1->track() <= e2->track();
+    }
+
+    // default case, use stacking order
+    return e1->z() <= e2->z();
+}
+
+std::vector<EngravingItem*> NotationInteraction::elementsNear(const mu::PointF& pos) const
```

Comment:
It turns out that this method _also_ already exists, namely under the name `NotationInteraction::hitElements`. (Shame on me for badly advising you initially on Discord 😅)